nucypher/crypto/kits.py

```diff
@@ -60,3 +60,38 @@ def __init__(self, *args, **kwargs) -> None:
     def from_bytes(cls, some_bytes):
         capsule, sender_pubkey_sig, ciphertext = cls.split_bytes(some_bytes)
         return cls(capsule=capsule, sender_pubkey_sig=sender_pubkey_sig, ciphertext=ciphertext)
+
+
+class RevocationKit:
+
+    def __init__(self, revocation_notices):
+        self.revocation_notices = revocation_notices
+        self.revocations = dict()
+
+    def __iter__(self):
+        return iter(self.revocations.items())
+
+    def __len__(self):
+        return len(self.revocation_notices)
+
+    @classmethod
+    def from_bytes(cls, some_bytes):
+        # TODO: Implement serialization
+        pass
+
+    def sign(self, signer):
+        """
+        Signs all the revocation notices provided in the kit and adds them to
+        the revocation dict ready for use.
+        """
+        for notice in self.revocation_notices:
+            notice.sign(signer)
```

Comment:
Can this return a bool, or does it raise an exception?

Reply:
It's not verification, it's just signing.